src/org/traccar/protocol/NavisProtocolDecoder.java

```diff
@@ -31,14 +33,26 @@
 import java.nio.charset.StandardCharsets;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Date;
 
 public class NavisProtocolDecoder extends BaseProtocolDecoder {
 
     private String prefix;
     private long deviceUniqueId, serverId;
+    private int flexDataSize;  // Bytes
```

Comment:
Why do we need instance variables?

Reply:
The FLEX protocol involves a negotiation procedure that specifies the bit mask of the fields and the size of this mask in bits. These data must be saved for correct parsing of further FLEX frames.
"flexDataSize" stores the size of the byte array for the bit  mask and can be calculated when parsing each frame, but I think it is not optimal.